frontend/src/metabase/dashboard/actions/save.js

```diff
@@ -117,10 +116,13 @@ export const updateDashboardAndCards = createThunkAction(
             name,
           })),
       });
-      dispatch(saveCardsAndTabs(updatedCardsAndTabs));
-
-      await dispatch(Dashboards.actions.update(dashboard));
 
+      // Make two parallel requests: one to update the dashboard and another for the dashcards and tabs
+      const [updatedCardsAndTabs] = await Promise.all([
+        updateCardsAndTabs,
+        dispatch(Dashboards.actions.update(dashboard)),
+      ]);
+      dispatch(saveCardsAndTabs(updatedCardsAndTabs));
```

Comment:
@calherries you changed the order, so now `dispatch(saveCardsAndTabs(updatedCardsAndTabs));` is after `dispatch(Dashboards.actions.update(dashboard));`, but before it was done in parallel

Reply:
ah, actually it should be fine as we update only FE data there, nvm

Reply:
Yes you're talking about `saveCardsAndTabs` I assume, which is handled [here](https://github.com/metabase/metabase/blob/6cec76055335c80092f3697c3011c8c8461916c3/frontend/src/metabase/dashboard/actions/tabs.ts#L282-L309). I will explain this context as a comment next time.

Reply:
Actually `dispatch(saveCardsAndTabs(updatedCardsAndTabs))` doesn't have to wait for `dispatch(Dashboards.actions.update(dashboard))` to be finished first.

If we want to optimize the waiting time we could do something like this
```ts
updateCardsAndTabs.then((updatedCardsAndTabs) => {
  dispatch(saveCardsAndTabs(updatedCardsAndTabs))
})

await Promise.all([
  updateCardsAndTabs,
  dispatch(Dashboards.actions.update(dashboard)),
])
```

This has duplications, but will perform faster on paper.

Reply:
Good point, though unfortunately it doesn't really make any difference in practice because the `updateCardsAndTabs` promise should take about the same time or longer than the `Dashboards.actions.update` dispatch. Anyway, I've updated it in case this changes.